File: assignments/week1/day1/problem5.java
// v) Write a program to find the smallest number divisible by all the numbers between 1 to n.


import java.util.Scanner;

class Solution{
    public static void main(String[] args){
        Scanner scn = new Scanner(System.in);

        long n = scn.nextLong();

        long ans = 1;
        for(long i = 2; i<n; i++){
            ans = lcm(ans, i);
        }

        System.out.println(ans);
    }

    public static long lcm(long a, long b){
        long min = Math.min(a,b);
        long max = Math.max(a,b);

        for(long i = max; ; i += max){
            if(i % min == 0) return i;
        }
    }
}

File: assignments/week1/day1/problem5.py
# v) Write a program to find the smallest number divisible by all the numbers between 1 to n.

def divisible_by_all(n:int):
    ans = 1
    for i in range(2, n+1):
        ans = lcm(ans, i)
    return ans

def lcm(a:int, b:int):
    min_val = min(a,b)
    max_val = max(a,b)

    count = max_val
    
    while True:
        if count % min_val == 0:
            return count
        count += max_val


print(divisible_by_all(5))